### PR TITLE
Review Guide

### LINKED ISSUE #132: Reviewing Pull Requests
As suggested by @sajayantony, it is good to [make a reviewing guide](https://contribute.cncf.io/maintainers/github/templates/recommended/reviewing/) for `oras` and `oras-go`.

Let's start with the `oras-go` repository, and do a copy in `oras` once the doc is revised.

### PR DESCRIPTION
**Pull request by shizhMSFT**
Resolves #132

### CODE REVIEW

File: REVIEWING.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+* **Transparency** - If a pull request won't be merged, clearly say why and close it. If a pull request won't be reviewed for a while, let the author know so they can set expectations and understand why it's blocked.
+* **Integrity** - Put the project's best interests ahead of personal relationships or company affiliations when deciding if a change should be merged.
+* **Stability** - Only merge when then change won't negatively impact project stability. It can be tempting to merge a pull request that doesn't meet our quality standards, for example when the review has been delayed, or because we are trying to deliver new features quickly, but regressions can significantly hurt trust in our project.
+
+
+## Process
+
+1. Do not start reviewing a pull request if it is WIP or is a draft pull request until it is marked as **ready to review**.
+2. Do not start reviewing a pull request if automated checks fail. Wait until they are fixed by the author.
+3. All pull requests except those generated by bots MUST have an issue associated. Motivations and designs SHOULD be discussed in the corresponding issue instead of in the pull request.
```
Comment (northtyphoon) [inline REVIEWING.md:42]: Is it for reviewer or author?

File: REVIEWING.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+4. Do a quick check for the license header for all new files.
+5. When you provide feedback, make it clear if the change must be made for the pull request to be approved, or if it is just a suggestion. Mark suggestions with `nit`.
+6. Reviews from maintainers are required. Only pull requests approved by maintainers can be merged. If the author is a maintainer, the pull request is required to be approved by other maintainers.
+7. Pull requests are required to be reviewed again if updates come after the approval from maintainers.
+
+
+## Checklist
+
+Below are a set of common questions that apply to all pull requests:
```
Comment (northtyphoon) [inline REVIEWING.md:51]: Do we expect reviewer to run the check list?

File: REVIEWING.md (hunk 1 above)
Comment (shizhMSFT) [reply REVIEWING.md:42]: The issue creation can be done anyone. For example, feature proposal or bug report.

File: REVIEWING.md (hunk 2 above)
Comment (shizhMSFT) [reply REVIEWING.md:51]: Yes, we expect both the author and the reviewer to run the checklist.

File: REVIEWING.md (hunk 1 above)
Comment (shizhMSFT) [reply REVIEWING.md:42]: > Is it for reviewer or author?

The answer is both.

File: REVIEWING.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+### Trust
+
+Be trustworthy. During a review, your actions both build and help maintain the trust that the community has placed in this project. Below are examples of ways that we build trust:
+
+* **Transparency** - If a pull request won't be merged, clearly say why and close it. If a pull request won't be reviewed for a while, let the author know so they can set expectations and understand why it's blocked.
+* **Integrity** - Put the project's best interests ahead of personal relationships or company affiliations when deciding if a change should be merged.
+* **Stability** - Only merge when then change won't negatively impact project stability. It can be tempting to merge a pull request that doesn't meet our quality standards, for example when the review has been delayed, or because we are trying to deliver new features quickly, but regressions can significantly hurt trust in our project.
+
+
+## Process
```
Comment (Wwwsylvia) [inline REVIEWING.md:38]: Can maintainers merge their own pull requests after it has been reviewed?  Or do they need to wait for other maintainers to merge it?

Comment (shizhMSFT) [reply REVIEWING.md:38]: Yes, they can merge if the PR is approved by other maintainers.

Comment (Wwwsylvia) [reply REVIEWING.md:38]: Should we add this into the process session?

**Comment by sajayantony** [comment]
@bridgetkromhout - would you be able to take a look at this?

File: REVIEWING.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+* If interactions with an author are not following code of conduct, close the PR and raise it up with your Code of Conduct committee or point of contact. It's not your job to coax people into behaving.
+
+### Trust
+
+Be trustworthy. During a review, your actions both build and help maintain the trust that the community has placed in this project. Below are examples of ways that we build trust:
+
+* **Transparency** - If a pull request won't be merged, clearly say why and close it. If a pull request won't be reviewed for a while, let the author know so they can set expectations and understand why it's blocked.
+* **Integrity** - Put the project's best interests ahead of personal relationships or company affiliations when deciding if a change should be merged.
+* **Stability** - Only merge when then change won't negatively impact project stability. It can be tempting to merge a pull request that doesn't meet our quality standards, for example when the review has been delayed, or because we are trying to deliver new features quickly, but regressions can significantly hurt trust in our project.
```
Comment (susanshi) [inline REVIEWING.md:35]: typo here from template "Only merge when _the_ change won't"

**Review by karenhchu — COMMENTED** [review]
One small edit. Otherwise it looks good!

File: REVIEWING.md (hunk 4 above)
Comment (shizhMSFT) [reply REVIEWING.md]: fixed typo

File: REVIEWING.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+Be welcoming and inclusive. You should proactively ensure that the author is successful. While any particular pull request may not ultimately be merged, overall we want people to have a great experience and be willing to contribute again. Answer the questions they didn't know to ask or offer concrete help when they appear stuck.
+
+### Sustainability
+
+Avoid burnout by enforcing healthy boundaries. Here are some examples of how a reviewer is encouraged to act to take care of themselves:
+
+* Authors should meet baseline expectations when submitting a pull request, such as writing tests.
+* If your availability changes, you can step down from a pull request and have someone else assigned.
+* If interactions with an author are not following code of conduct, close the PR and raise it up with your Code of Conduct committee or point of contact. It's not your job to coax people into behaving.
```
[the remark this answers is not in the document]
Comment (shizhMSFT) [reply REVIEWING.md]: fixed grammar

**Comment by shizhMSFT** [comment]
Pinging @jdolitsky @deitch @sajayantony for review.

**Review by sajayantony — APPROVED** [review]
Let' start with this for now.

**Review by sajayantony — APPROVED** [review]

**Pull request merged**